### PR TITLE
Comments: Janitorial Cleanup of CommentsList

### PR DESCRIPTION
**Pull request by samouri**
prettify + e6nextify
There should be no functional changes

- removed usage of `bindActionCreators` in favor of object shorthand syntax
- moved propTypes, defaultProps, initialState to the esnexty way of doing it
- removed manual binds in favor of the ` = () => ` syntax

I'm doing this to set the stage for bigger changes later today/tomorrow.

To Test:
run through all of the comments-related actions on comments list.  make sure they all still work

- [ ] onEditCommentClick
- [ ] onEditCommentCancel
- [x] onReplyClick
- [x] onReplyCancel

etc

### CODE REVIEW

**Comment by samouri** [comment]
In order to stay sane, i recommend looking at the commits individually

**Comment by blowery** [comment]
į̵̛̙̳̰̮̮̟͉͇̫͊̿̀̃͑́ l̷̗̜̻̤̰͙̣̗̒͆̈͆̒͌͢ȏ̴̢̧̙͍̤̘̟̃͑̀̊̌̿ơ̢̝͉̺̯̩̌͛̆̓́͊̑̇͝ķ̸͈̝̠̱̤̤̻̙̩͌͊́̓̓̂̑͂e̷̪͇̞̝͔͐̈́̉̍̀́̄̽͌̚ͅd̼͖͚̙͙̲̜̠̜̝̾͆̒͗́͌͗̀̈́ ą̷̢͚̪̻̭͆̒̈͗̋̚ț̶̨̛̜̯̜̗̰̓̇̒͆̆̓͜͞͝ ṫ̙̮̤̻̭́͊̂̆͌̂̒͜͞ͅh̵̹̙͓̹̮̳̃̑͆̄́̃͘͝͞ͅe̴̬̼̪̪̬͒͑̀̋͌͋͝ w̵̗̻̻̲̲̜̍̄̓̈́̎̔͘͢͢͜͞͠͡h̨̭̖̬̘̼̤͎͛̔̔͑̿̎̎̕͠o͓̝̺̯̼̻͎̘͆̃̂̇͐̿͛̐͌͞ľ̸̝̰̱͍͍͕͉̣̜͙̋̀̈́́̽̑̅̕e̸͔̼̞̻̗̖͚͒̄͐̕͝͞ d̷͙̻̱̭̻̥̿͋́̅͐̓̍͠i̧̧̖͇͚̟̓͌̿̇̚͢f̷̲̠͙̝̟̹͔̞̍̃̇̄̀͗̽̓̊̚f̵̻͈͙̭͙̾̈̈́̌̐͐͟͟.̸͚̪̥͚́͐̾̐̾͗̓͟͡
̸̧̣̟̻̹̫̮̤̠̮̓̃̐͘͘͝͠
̷̛̺̘͔͈̳̭̰͑̂́͌̒̕͜͢ͅą̛̰̣̦͔̍̔̎̈́̌ͅņ̴̦̮͔̳̲̉̿̅̊̽̎̽͜d̶̢̗̼̹̟͇̣̙̠͚̅̽͐̾͐̚͡ i̢͔͙̖͖̋͛́͛̀͆̚̚͟͟t̵̞͔̼̪͇͈͍̍͑͛͌͋͟͝ ľ̸̨̛͉̯̺̻̫͓̙̜͂̿́͌̑͜͡o̡̧͔̼̞͉̤͕̊̃͒̋̎̋̉͒̕͟ǫ̶̯͍͍͋͑̄̽̉͊̆͟k̵̻̰̖̞͙͌̇̈͛̄́͛͌͢͟ȩ̵̧̦̪͖͇̗̣̒̽͂̐̈̅͟d̶̝̱̙̘̾̌̉̾̂̿̀͆͟͞ b̯̖̺̝̙͍̤̥͈̩̊̽̔̒͋̾̾͗̉ä̪̼͉͖̟͛͛́͜͠c̰̲̺̲̯͇͒͌̀̃͗́̊̈̏̚ͅk̵̢͔̻̞͕̈̒͐̒̚͡.̛̘̯̙̤͇͈̠͑̏͆̀̌͊͞

File: client/blocks/comments/post-comment-list.jsx [hunk 1]
```diff
@@ -60,111 +70,113 @@ class PostCommentList extends React.Component {
...
 		recordGaEvent( 'Clicked Reply to Comment' );
 		recordTrack( 'calypso_reader_comment_reply_click', {
 			blog_id: this.props.post.site_ID,
-			comment_id: commentID
+			comment_id: commentID,
 		} );
 	}
 
-	onReplyCancel() {
+	onReplyCancel= () => {
```
Comment (blowery) [inline client/blocks/comments/post-comment-list.jsx]: missing space?

Comment (samouri) [inline client/blocks/comments/post-comment-list.jsx]: forgot to repretty after es6nexting. fixed now

**Comment by blowery** [comment]
hmmm... do we actually expose edit comment? 

oh, on my sites i bet...

**Comment by samouri** [comment]
> oh, on my sites i bet...

nailed it :)

**Comment by blowery** [comment]
edit comment seems to edit the wrong comment on local dev. Interesting, the edit link doesn't appear on wpcalypso / staging / production. checking master on local to see if it works there.

**Comment by blowery** [comment]
broken on master too. :/
edit-dem-comments

**Comment by samouri** [comment]
Thats no good. submitted an issue here: https://github.com/Automattic/wp-calypso/issues/16212

**Pull request merged**